django/http/request.py

```diff
@@ -347,12 +417,12 @@ def __init__(self, query_string=None, mutable=False, encoding=None):
                 except UnicodeDecodeError:
                     # ... but some user agents are misbehaving :-(
                     query_string = query_string.decode('iso-8859-1')
-            for key, value in parse_qsl(query_string or '',
+            for key, value in parse_qsl(SplitWrapper(query_string or '', limit_checker),
```

Comment:
If someone with write access to this repository tells me that there is a higher chance of this PR being merged by importing parse_qsl()'s code I will do it. Otherwise I feel I'm just wasting my time here. The patch works for me.